### PR TITLE
MGDSTRM-10039 elevating the priority of broker and zookeeper pods

### PR DESCRIPTION
**Pull request by shawkins**
also adding some trivial instance profiler changes.

### CODE REVIEW

**Comment by shawkins** [comment]
I'm not sure what a good naming convention is here.  We already have kas-fleetshard-high.  I went with core, but you could also go with standard, moderate, medium high, etc.

The choice of 10 for the priority value is also arbitrary.  We can of course space that out more if desired - high is 1000000

**Review by MikeEdgar — APPROVED** [review]
lgtm. I'm not sure about the naming either. I don't object to using `core`. Maybe something like `operand-high` or similar would be more descriptive?

**Comment by rareddy** [comment]
Since "high" name already set precedent, I like the @shawkins idea of going with "standard, moderate, medium high, etc." then there is no ambiguity when referred in conjunction with other class types.

File: operator/src/test/java/org/bf2/operator/operands/KafkaClusterTest.java [hunk 1]
```diff
@@ -216,6 +216,18 @@ private void alternativeConfig(Consumer<KafkaInstanceConfiguration> configModifi
...
 
+    @Test
+    void testElevatedPriority() throws IOException {
+        final ManagedKafka managedKafka = exampleManagedKafka("2Ti");
+
+        Mockito.when(overrideManager.useElevatedPriority(Mockito.anyString())).thenReturn(true);
+
+        Kafka kafka = kafkaCluster.kafkaFrom(managedKafka, null);
+
+        assertEquals("kas-fleetshard-core", kafka.getSpec().getKafka().getTemplate().getPod().getPriorityClassName());
```
Comment (rareddy) [inline operator/src/test/java/org/bf2/operator/operands/KafkaClusterTest.java:227]: I guess these need to be updated to new name to fix the unit tests

Comment (shawkins) [reply operator/src/test/java/org/bf2/operator/operands/KafkaClusterTest.java]: Yes, sorry things were incomplete.  Should be good now.

**Review by rareddy — APPROVED** [review]
lgtm

**Pull request merged**